superset-frontend/packages/superset-ui-chart-controls/src/components/ColumnOption.tsx

```diff
@@ -48,14 +52,27 @@ export function ColumnOption({
   const hasExpression = expression && expression !== column_name;
   const type = hasExpression ? 'expression' : type_generic;
   const [tooltipText, setTooltipText] = useState<ReactNode>(column.column_name);
+  const [columnTypeTooltipText, setcolumnTypeTooltipText] = useState<ReactNode>(
+    column.type,
+  );
 
   useLayoutEffect(() => {
     setTooltipText(getColumnTooltipNode(column, labelRef));
+    setcolumnTypeTooltipText(getColumnTypeTooltipNode(column));
   }, [labelRef, column]);
 
   return (
     <StyleOverrides>
-      {showType && type !== undefined && <ColumnTypeLabel type={type} />}
+      {showType && (
```

Comment:
```suggestion
      {showType && type !== undefined && (
```
Should we add this check here?

Reply:
> Really appreciate this fantastic PR, particularly the quick turnaround on suggestions (both code and design), and clear communication. You even added tests, and managed to make CI happy! This is a very well-done first PR, and I hope we see more like it in the future :) I added a couple more (non-blocking) suggestions, but I'm happy to approve it as-is, and look forward to merging this!

Thanks for suggesting changes @rusackas ! I have included your review comments now in PR.

Reply:
Yes, I have included your review comments in recent commit.